examples/sagemaker_step_resource/README.md

```diff
@@ -0,0 +1,97 @@
+This example shows how you can use the `StepOperator` class to run your training jobs on Sagemaker.
+
+Sagemaker offers specialised compute instances to run your training jobs and has a beautiful UI to track and manage your models and logs. You can now use ZenML to submit individual steps to be run on compute instances managed by Amazon Sagemaker. 
+
+## Overview
+Here we train a simple sklearn classifier on the MNIST dataset using Amazon Sagemaker.
+
+## Run it locally
+
+Currently, step operators only work with a local orchestrator but support for cloud orchestrators is on the way soon!
+
+### Installation
+In order to run this example, you need to install and initialize ZenML and the necessary integrations:
+
+```shell
+# install CLI
+pip install zenml
+
+# install ZenML integrations
+zenml integration install aws sklearn 
+
+# pull example
+zenml example pull sagemaker_step_resource
+cd zenml_examples/sagemaker_step_resource
+
+# initialize
+zenml init
+```
+
+### Pre-requisites
+
+In order to run the example, you need to setup a few things to allow ZenML to interact with Sagemaker.
+
+* First, you need to create a role in the IAM console that you want the jobs running in Sagemaker to assume. This role should at least have the `AmazonS3FullAccess` and `AmazonSageMakerFullAccess` policies applied. Check [this link](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-create-execution-role) to learn how to create a role.
+
+* Next, you need to choose what instance type needs to be used to run your jobs. You can get the list [here](https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html).
+
+* Optionally, you can choose an S3 bucket to which Sagemaker should output any artifacts from your training run. 
+
+* You can also supply an experiment name if you have one created already. Check [this guide](https://docs.aws.amazon.com/sagemaker/latest/dg/experiments-create.html) to know how. If not provided, the job runs would be independent.
+
+* You can also choose a custom docker image that you want ZenML to use as a base image for creating an environment to run your jobs in Sagemaker. 
+
+* You need to have the `aws` cli set up with the right credentials. Make sure you have the permissions to create and manage Sagemaker runs. 
+
+* A container registry has to be configured in the stack. This registry will be used by ZenML to push your job images that Sagemaker will run. Check out the [cloud guide](https://docs.zenml.io/features/cloud-pipelines/guide-aws-gcp-azure) to learn how you can set up an elastic container registry. 
+
+Once you have all these values handy, you can proceed to setting up the components required for your stack.
+
+### Creating the stack
+
+The stack will consist of:
+- The local metadata store
+- The local orchestrator
+- An S3 artifact store
+- The Sagemaker step operator
+
+```bash
+zenml artifact-store register s3-store \
+    --type s3
```

Comment:
The options when registering the artifact store/step operator must be of the form --key=value, the `=` here is important as I think otherwise the parsing fails

Reply:
Ah, interesting! I also got an error when registering the step operator but didn't know why. 
![image](https://user-images.githubusercontent.com/37150991/157869676-30c92630-319d-4bd3-9f28-e8cf488dc31e.png)
I'll fix these